### PR TITLE
Edit OAuth client screen

### LINKED ISSUE #437: [Portal] Layout Edit OAuth Client Screen
- [x] Reuse modify OAuth client form
- [x] Add client ID field, read-only on top
- [x] pre-fill form with existing setting

### PR DESCRIPTION
**Pull request by axolotl0212**
base on #434

Closes #437

- reuse modify oauth client form

# Screenshot

Screenshot 2020-10-06 at 4 56 51 PM

### CODE REVIEW

File: portal/src/graphql/portal/EditOAuthClientScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+      .then((result) => {
+        if (result != null) {
+          setSubmittedForm(true);
+        }
+      })
+      .catch(() => {});
+  }, [clientConfig, updateAppConfig, rawAppConfig]);
+
+  useEffect(() => {
+    if (submittedForm) {
```
Comment (louischan-oursky) [inline portal/src/graphql/portal/EditOAuthClientScreen.tsx:89]: It seems that the only use of `submittedForm` is here. Why don't we just call `navigate` in line 82? Using an useEffect in this way sounds a bit indirect.

**Review by louischan-oursky — COMMENTED** [review]
The client ID is not editable, which is correct. But it is not visually read-only. https://developer.microsoft.com/en-us/fluentui#/controls/web/textfield I see Fluent does have a disabled state. Can we use it?

**Comment by axolotl0212** [comment]
> The client ID is not editable, which is correct. But it is not visually read-only. https://developer.microsoft.com/en-us/fluentui#/controls/web/textfield I see Fluent does have a disabled state. Can we use it?

Sure, can add disabled={true}. I used readOnly props simply because this is what we want to achieve. btw, using disabled the text will have less contrast.

File: portal/src/graphql/portal/EditOAuthClientScreen.tsx (hunk 1 above)
Comment (axolotl0212) [reply portal/src/graphql/portal/EditOAuthClientScreen.tsx:89]: This is for making sure the state is set before calling navigate, as `setState` in functional component does not have callback @louischan-oursky

**Comment by axolotl0212** [comment]
After using disabled props, it looks like this


Screenshot 2020-10-07 at 7 28 45 PM

**Comment by louischan-oursky** [comment]
I just merged the add-`name` to client PR. Can you add support for it? Right now I cannot create new client :(

> After using disabled props, it looks like this

Maybe we should just render it with `Text`?

File: portal/src/graphql/portal/EditOAuthClientScreen.tsx (hunk 1 above)
Comment (louischan-oursky) [reply portal/src/graphql/portal/EditOAuthClientScreen.tsx:89]: A second thought. I guess we should stay on the page even after saving.

**Comment by axolotl0212** [comment]
> I just merged the add-`name` to client PR. Can you add support for it? Right now I cannot create new client :(

Sure, I just rebased, can quickly change it.

**Comment by axolotl0212** [comment]
Render as `Text`

Screenshot 2020-10-07 at 7 43 04 PM

**Comment by axolotl0212** [comment]
updated and added support for name field in config

**Comment by louischan-oursky** [comment]
Right now when I click add client, the app will crash with calling `.trim()` on undefined. I guess this is related to our usage of `!`.

**Comment by axolotl0212** [comment]
@louischan-oursky Fixed the problem, I forgotten to pass client name to the ModifyOAuthClientForm. Changed to `?? ""` to make it safer

**Pull request merged**